File: leetcode/medium/1769. Minimum Number of Operations to Move All Balls to Each Box/solution.cc
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<int> minOperations(string boxes) {
        int size = boxes.size();
        vector<int> vec(size, 0);

        int sum = 0, exist = 0;
        for (int i = 0; i < size; i++) {
            vec[i] += sum;
            if (boxes[i] == '1') exist += 1;
            sum += exist;
        }

        sum = 0, exist = 0;
        for (int i = size - 1; i >= 0; i--) {
            vec[i] += sum;
            if (boxes[i] == '1') exist += 1;
            sum += exist;
        }

        return vec;
    }
};

int main() {
    Solution solution;
    vector<int> s = solution.minOperations("110");
    for (int i = 0; i < s.size(); i++) {
        cout << s[i] << endl;
    }
    return 0;
}

File: leetcode/medium/1769. Minimum Number of Operations to Move All Balls to Each Box/solution.js
// https://leetcode.com/problems/minimum-number-of-operations-to-move-all-balls-to-each-box/

/**
 * @param {string} boxes
 * @return {number[]}
 */
function minOperations(boxes) {
    const len = boxes.length;
    const counter = Array(len).fill(0);

    let sum = 0, exist = 0;    
    for (let i = 0; i < len; i += 1) {
        counter[i] += sum;
        if (boxes[i] === '1') exist += 1;
        sum += exist;
    }

    sum = 0, exist = 0;
    for (let i = len - 1; i >= 0; i -= 1) {
        counter[i] += sum;
        if (boxes[i] === '1') exist += 1;
        sum += exist;
    }

    return counter;
};

console.log(minOperations("001011"));